### ISSUE TITLE
An add-on Disabled by Mozilla has an active New Version link in 'Manage my submissions' page

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-server#7383]
STR:

Prerequisites:  user has admin reviewer permissions

1. Log into Developer Hub and submit a new extension
2. Wait for the add-on to pass auto approval, 
3. Go to Reviewer Tools and search for your add-on in the auto-approved queue
4. In the add-on review page, scroll down to More Actions section and click on 'Force disable Add-on'
5. Go back to DevHub -> Manage my submissions and look for the disabled add-on
6. Click on the New Version link present under the the add-on name

Actual result:
The New Version link is active and is redirecting the user on a 'not found' page

Expected result:
The New Version link is disabled when an add-on is Disabled by Mozilla

Notes:
Reproduced on AMO dev and stage with FF58 (Win10x64)

![new v](https://user-images.githubusercontent.com/31961530/35337205-42e3c0c4-0123-11e8-88b7-448779267c8f.gif)

**Comment by diox** [comment]
I don't think it's a new issue, as I haven't changed this with #7282 but it'd be nice to fix it.

**Comment by AlexandraMoga** [comment]
Verifed fixed on AMO-dev FF58 (Win10)

![image](https://user-images.githubusercontent.com/31961530/35858690-17e03812-0b46-11e8-84c2-9bb7534f818d.png)
